deploy/cdk/lib/manifest-pipeline/manifest-pipeline-stack.ts

```diff
@@ -435,6 +441,42 @@ export class ManifestPipelineStack extends Stack {
       })
     }
 
+    if ((props.createCopyMediaContentLambda !== undefined) && props.createCopyMediaContentLambda) {
+
+      const marbleContentBucket = Bucket.fromBucketName(this, 'MarbleContentBucket', props.marbleContentBucketName)
+
+      const copyMediaContentLambda = new Function(this, 'CopyMediaContentLambda', {
+        code: AssetHelpers.codeFromAsset(this, path.join(props.lambdaCodeRootPath, 'copy_media_content/')),
+        description: 'Copies media files from other folders to /public-access/media folder to be served by CDN',
+        handler: 'handler.run',
+        runtime: Runtime.PYTHON_3_8,
+        environment: {
+          SENTRY_DSN: props.sentryDsn,
+        },
+        timeout: Duration.seconds(900),
+        initialPolicy: [
+          new PolicyStatement({
+            effect: Effect.ALLOW,
+            actions: ["storagegateway:ListFileShares"],
+            resources: [Fn.sub('arn:aws:storagegateway:${AWS::Region}:${AWS::AccountId}:*')],
+          }),
+          new PolicyStatement({
+            effect: Effect.ALLOW,
+            actions: ["storagegateway:DescribeSMBFileShares",
+              "storagegateway:RefreshCache",
+            ],
+            resources: [Fn.sub('arn:aws:storagegateway:${AWS::Region}:${AWS::AccountId}:share/*')],
+          }),
```

Comment:
I'm a little concerned with giving this permission to do this on all gateways and all shares. Looking at https://docs.aws.amazon.com/service-authorization/latest/reference/list_amazonstoragegateway.html#amazonstoragegateway-share, you should be able to specify a gateway id for the ListFileShares and a share id for the RefreshCache/DescribeSMBFileShares actions. I'd recommend limiting this to a gateway id and share id, passed in via context, SSM, or a CloudFormation Import. Maybe you already have these defined somewhere?

Reply:
I don't have these defined.  My lambda function actually accepts the bucket name, does a search to find all file shares, then for each of those, it gets the smb shares until it find the arn containing the bucket-name passed.  It then does a refresh_cache on that arn.

I assumed this would be safe since I'm just reading listings, and don't have any other privileges for this.

I'll take a look at your link tomorrow.